index.js

```diff
@@ -1,46 +1,113 @@
     const tableElements = $('.table-responsive table tr').toArray();
+
+    //init counter and content for loop
     let i = 0;
     const content = {};
-    tableElements.forEach(element => {
+
+    //for each tr
+    tableElements.forEach(tr => {
+        //increment counter
         i++;
+
+        //add array for current table row
         if (!content[i]) content[i] = [];
-        element.children.forEach(children => {
+
+        //for each td
+        tr.children.forEach(td => {
+            //init child content
             let childContent = '';
-            if (children.children) children.children.forEach(c => {
+
+            //for each td child as c (different translation of word)
+            if (td.children) td.children.forEach(c => {
+                //add c to content string
                 if (c.data) childContent = childContent + `${c.data}|`;
             });
+
+            //push td content string to content array
             content[i].push(childContent);
         });
     });
 
+    //init result
     const returnForm = {};
 
+    //return if content does not exist
     if (!content[1]) return null;
 
+    //init counter
     let y = 0;
+
+    //for each td content string from first tr as c
     content[1].forEach(c => {
-        if (content[3][y]) returnForm[translations[c.split('|').join('')] ? translations[c.split('|').join('')] : c.split('|').join('')] = content[3][y].substring(0, content[3][y].length - 1);
-        else returnForm[translations[c.split('|').join('')] ? translations[c.split('|').join('')] : c.split('|').join('')] = content[2][y].substring(0, content[2][y].length - 1);
+        //get keys
+        var key_returnForm = attr_translations[c.split('|').join('')] ? attr_translations[c.split('|').join('')] : c.split('|').join('');
+        var key_content = content?.[3]?.[y] ? 3 : 2;
+
+        //get y-th translation
+        returnForm[key_returnForm] = content[key_content][y].substring(0, content[key_content][y].length - 1);
+
+        //increment
         y++;
     });
+    
+    //add word to result array
     returnForm['latin'] = word;
+    
+    //delete element with empty key
     delete returnForm[''];
 
+    //return result
     return returnForm;
-};
\ No newline at end of file
+};
+
+
+module.exports.translate = async function (sentence, callback) {
+    //init result object
+    var res = {}
+    
+    //prepare sentence as an array
+    sentence = sentence.split(',').join('').split('.').join('').split(' ')
+    
+    //get sentence word length
+    var length = sentence.length
+    
+    //loop through words
+    sentence.forEach(async (w, key) => {
+        //get word info
+        var wordInfo = await module.exports.getWordInformation(w) ? await module.exports.getWordInformation(w) : w
+    
+        //put word info into result array
+        res[key] = wordInfo
+    
+        //print object when all translations fetched
+        if(Object.keys(res).length == length) await callback(res)
```

Comment:
```suggestion
        if(Object.keys(res).length === length) await callback(res)
```
=== needs to be used here, [learn more](https://howtodoinjava.com/javascript/javascript-equality-vs-identity-operators/).

Reply:
and why use === in this context? in my opinion it doesn't matter....

Reply:
In this situation it this not really matter, but it's best practise to use `===` instead of `==` unless you actually *need* to use `==`. This is the case, because `==` first tries to parse both values to the same type which can introduce bugs depending on the context.

Reply:
I had flaged it, because I am used to find and flag issues like this in PRs, sorry 😅